src/subdomains/generic/kyc/services/integration/financial.service.ts

```diff
@@ -10,8 +11,11 @@ import { QuestionType } from '../../enums/kyc.enum';
 export class FinancialService {
   constructor(private readonly i18n: I18nService) {}
 
-  getQuestions(lang: string = Config.defaults.language.toLowerCase()): KycFinancialQuestion[] {
-    return FinancialQuestions.map((q) => ({
+  getQuestions(
+    lang: string = Config.defaults.language.toLowerCase(),
+    accountType: AccountType,
+  ): KycFinancialQuestion[] {
+    return FinancialQuestions.filter((q) => q.accountTypes.includes(accountType)).map((q) => ({
       key: q.key,
```

Comment:
Maybe adapt the filter like this
```ts
 return FinancialQuestions.filter((q) => !q.accountTypes || q.accountTypes.includes(accountType)).map((q) => ({ ...
```
and remove the accountTypes array from entries which are active for all?